job-server/src/spark.jobserver/JobServer.scala

```diff
@@ -26,6 +26,13 @@ import org.slf4j.LoggerFactory
 object JobServer {
   val logger = LoggerFactory.getLogger(getClass)
 
+  private def setSystemPropertiesFromConfig(config: Config) {
```

Comment:
I think we should avoid adding code to JobServer.scala as this is difficult to test as @velvia said.